### PR TITLE
add some explanations on how to build docs to fix #2309

### LINKED ISSUE #2309: Explain how to build the docs
**Is your feature request related to a problem? Please describe.**

There is no documentation explaining how to build the docs for Cutter

**Describe the solution you'd like**

A dedicated page under `docs/contributing/docs/` that will instruct how to build Cutter's docs. Explain the requirements and the different `make` commands such as `make html` and `make quick`.

**Describe alternatives you've considered**

To add it to the `getting-started` document under `docs/contributing/docs`.

### PR DESCRIPTION
**Pull request by kohnakagawa**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [ ] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
I added the explanation on how to build docs for Cutter to fix #2309.

**Test plan (required)**
Build docs and check the output.

**Closing issues**
"closes #2309"

### CODE REVIEW

File: docs/source/contributing/docs.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Contributing Documentation
+==========================
+
+.. toctree::
+   :maxdepth: 2
+   
+   Getting Started <docs/getting-started>
+   docs/building-docs
```
Comment (ITAYC0HEN) [inline docs/source/contributing/docs.rst:8]: It's a little hard to find and access the building-docs page. The only option I can see is that the user will go to

Contributing -> click the DOCS button -> see that there is a "building docs" in the sidebar


What I suggest is to also mention the building docs page from the getting-started page

**Comment by kohnakagawa** [comment]
@ITAYC0HEN  Sorry for the late response. I fix the source of documentation according to your review.
Is it necessary to squash these commits? If so, I will handle this.

File: docs/source/contributing/docs/building-docs.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Building docs
 =======================
 
-This page contains how to build this Cutter documentation.
+This page explains the step that are needed to build Cutter's documentation.
```
Comment (xarkes) [inline docs/source/contributing/docs/building-docs.rst:4]: steps? plural?

Comment (kohnakagawa) [reply docs/source/contributing/docs/building-docs.rst]: Thank you! I changed this from "step" to "steps".

**Comment by kohnakagawa** [comment]
Thank you. I fix sub-bullets.

スクリーンショット 2020-08-15 14 50 55

**Pull request merged**